packages/node-cli/src/commands/project/deploy/DeployAction.js

```diff
@@ -70,17 +75,51 @@ module.exports = class DeployAction extends BaseAction {
 				flags.push(COMMAND.FLAGS.APPLY_CONTENT_PROTECTION);
 			}
 
-			const sdkParams = CommandUtils.extractCommandOptions(params, this._commandMetadata);
+			if (params[COMMAND.FLAGS.PREVIEW]) {
+				try {
+					delete params[COMMAND.FLAGS.PREVIEW];
+					flags = flags.slice((0, 2));
+
+					if (flags.includes(COMMAND.FLAGS.VALIDATE)) {
+						throw NodeTranslationService.getMessage(COMMAND_DEPLOY.ERRORS.VALIDATE_AND_DRYRUN_OPTIONS_PASSED);
+					}
+
+					const sdkParams = CommandUtils.extractCommandOptions(params, this._commandMetadata);
+
+					const executionContextForDryrun = SdkExecutionContext.Builder.forCommand(previewCommand)
```

Comment:
I would use camelCase also in the "run" : executionContextForDryRun

Reply:
I interpreted dry run as one word, so I didn't camel cased it.

Reply:
ok, I thought about it because the flag is -dry-run but it's ok